src/parthenon_arrays.hpp

```diff
@@ -0,0 +1,309 @@
+//========================================================================================
+// Parthenon++ astrophysical MHD code
+// Copyright(C) 2014 James M. Stone <jmstone@princeton.edu> and other code contributors
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef PARTHENON_ARRAYS_HPP_
+#define PARTHENON_ARRAYS_HPP_
+
+// C headers
+#include<assert.h>
+
+// C++ headers
+#include <cstddef> // size_t
+#include <string>
+#include <type_traits>
+#include <utility> // make_pair
+#include <vector>
+
+// Kokkos Headers
+#include <Kokkos_Core.hpp>
+
+// Parthenon++ headers
+#include "kokkos_abstraction.hpp"
+
+// Macro for automatically creating a useful name
+#define PARARRAY_TEMP\
+  "ParArrayNDGeneric:"+std::string(__FILE__)+":"+std::to_string(__LINE__)
+
+namespace parthenon {
+
+#define SLC0 std::make_pair(0,1)
```

Comment:
Could you perhaps move the definition of this closer to where it's first used, and then immediately undefine it after it's used? That makes it clearer that it's supposed to be temporary.

Reply:
Yep. Good idea.